src/main/java/org/deepsymmetry/Wayang.java

```diff
@@ -265,4 +276,51 @@ public static void sendFrame() {
             }
         }
     }
+
+    /**
+     * Send a frame of pixels async, corresponding to whatever has been drawn in the image returned by open(),
+     * to the display.
+     *
+     * @throws LibUsbException       if there is a problem communicating.
+     * @throws IllegalStateException if the Push 2 has not been opened.
+     */
+    public static void sendFrameAsync() {
+        if (transferBuffer == null) {
+            throw new IllegalStateException("Push 2 device has not been opened");
+        }
+        IntBuffer transferred = IntBuffer.allocate(1);
+        transfer = LibUsb.allocTransfer();
+        LibUsb.fillBulkTransfer(transfer, pushHandle, (byte) 0x01, headerBuffer, LibUsb::freeTransfer,  null, 1000);
+
+        int result = LibUsb.submitTransfer(transfer);
+        if (result != LibUsb.SUCCESS) {
+            throw new LibUsbException("Transfer of frame header to Push 2 display failed", result);
+        }
+
+        // We send 8 lines at a time to the display; allocate buffers big enough to receive them,
+        // expand with the row stride padding, and mask with the signal shaping pattern.
+        short[] pixels = new short[LINES_PER_TRANSFER * DISPLAY_WIDTH];
+        byte[] maskedChunk = new byte[LINES_PER_TRANSFER * BYTES_PER_LINE];
+        for (int i = 0; i < (DISPLAY_HEIGHT / LINES_PER_TRANSFER); i++) {
+            displayImage.getRaster().getDataElements(
+                    0, i * LINES_PER_TRANSFER, DISPLAY_WIDTH, LINES_PER_TRANSFER, pixels);
+            maskPixels(pixels, maskedChunk);
+            transferBuffer.clear();
+            transferBuffer.put(maskedChunk);
+            transferred.clear();
+
+            transfer = LibUsb.allocTransfer();
+            LibUsb.fillBulkTransfer(transfer, pushHandle, (byte) 0x01, transferBuffer, LibUsb::freeTransfer, null, 1000);
+            result = LibUsb.submitTransfer(transfer);
+
+            if (result != LibUsb.SUCCESS) {
+                throw new LibUsbException("Transfer of frame header to Push 2 display failed", result);
+            }
+        }
+
+        // handle events
+        result = LibUsb.handleEventsTimeout(null, 250000);
```

Comment:
Rather than assuming the events are ready to handle here, we should check to see if the event handling thread is already running; if not, we could start it here. If it is running, we don’t need to do anything here.